### PR TITLE
Extending XmlElement with ChildTexts and handling edge cases in ChildText

### PR DESCRIPTION
**Pull request by sharmi**
In XmlElement,
            One can query text of the first matching child using ChildText. I have added the ChildTexts function to query multiple elements. Also ChildText fails if there is no element matching the xpath and the selected child becomes nil. So I have added a check to see if ChildText is nil or not.

I have added the relevant tests for both items.

### CODE REVIEW

File: colly.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ import (
 	"github.com/kennygrant/sanitize"
 	"github.com/temoto/robotstxt"
 
-	"github.com/gocolly/colly/debug"
-	"github.com/gocolly/colly/storage"
+	"github.com/sharmi/colly/debug"
```
Comment (asciimoo) [inline colly.go]: Please don't change import urls.

**Comment by asciimoo** [comment]
Also, there are some minor formatting bugs reported by travis: https://travis-ci.org/gocolly/colly/jobs/352689219 . Please fix those, too.

**Comment by sharmi** [comment]
Hi asciimoo, 
      Thank you for the input. I will fix the formatting bugs. As far as the importing goes, I am in a fix. I am a golang novice, so I am still learning the ropes and any help would be appreciated.

When I fork the repo, and start using it as github.com/sharmi/colly in the crawler, the crawler fails to run as some of the files in github.com/sharmi/colly import github.com/gocolly/colly .  The crawler refused to run until I changed all the imports. When I don't change  the inputs and use colly in crawler, this is the error I get.

`./crawler.go:39:37: cannot use "github.com/sharmi/colly/debug".LogDebugger literal (type *"github.com/sharmi/colly/debug".LogDebugger) as type "github.com/gocolly/colly/debug".Debugger in argument to colly.Debugger:
        *"github.com/sharmi/colly/debug".LogDebugger does not implement "github.com/gocolly/colly/debug".Debugger (wrong type for Event method)
                have Event(*"github.com/sharmi/colly/debug".Event)
                want Event(*"github.com/gocolly/colly/debug".Event)
`

I did not want to trouble you too much with this. So I tried looking into a few other golang libraries like goquery but they did not have the subrepos like colly/debug. So I am unsure on how to handle this. Any help on how to resolve this would be great.

**Comment by kvch** [comment]
You have to checkout `gocolly/colly` to `$GOPATH/src/github.com/gocolly/colly`, because this is the upstream repo.
You should put repos under your github username if it is your project. If others forked your repo, they would put it under `sharmi/projectname`. So the imports don't have to be changed.

**Comment by sharmi** [comment]
@kvch Thank you so much. 

I tried googling and couldn't get any solution. Right then, I knew it would be a solution of simple common sense. Yet it eluded me for so long :( . Now I know. I am enjoying learning go and people like you make it so. Thanks again.

**Comment by asciimoo** [comment]
@sharmi any update on this? Let me know if you need further help.

**Comment by sharmi** [comment]
Sorry asciimoo, I got caught up in another project, So I took some time to fix this one. Hope the changes are as you requested.

File: xmlelement_test.go [hunk 2]
```diff
@@ -1,11 +1,11 @@
-package colly_test
+package colly
```
Comment (asciimoo) [inline xmlelement_test.go]: Please keep the package name `colly_test`

**Pull request merged**